ql/src/java/org/apache/hadoop/hive/ql/parse/UpdateDeleteSemanticAnalyzer.java

```diff
@@ -145,11 +163,38 @@ private void reparseAndSuperAnalyze(ASTNode tree, Table mTable, ASTNode tabNameN
       where = (ASTNode)children.get(whereIndex);
       assert where.getToken().getType() == HiveParser.TOK_WHERE :
           "Expected where clause, but found " + where.getName();
-    }
 
-    // Add a sort by clause so that the row ids come out in the correct order
-    appendSortBy(rewrittenQueryStr, columnAppender.getSortKeys());
+      if (shouldOverwrite) {
+        if (where.getChildCount() == 1) {
+
+          // Add isNull check for the where clause condition, since null is treated as false in where condition and
+          // not null also resolves to false, so we need to explicitly handle this case.
+          ASTNode functionNode = new ASTNode(new CommonToken(HiveParser.TOK_FUNCTION, "TOK_FUNCTION"));
```

Comment:
should we do this if `where.getChild(0)` is a terminal node?

Reply:
the end result of the tree can also be Unknown, and we need to safgaurd not inverting the unknown,

Reply:
👍